core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -367,7 +367,7 @@ public static InputStream getInputStream(URL url) throws IOException {
      * @since TODO
      */
     public static HttpClient newHttpClient() {
-        return newHttpClientBuilder().build();
+        return newHttpClientBuilder().followRedirects(HttpClient.Redirect.NORMAL).build();
```

Comment:
Unrelated change while I was here: while testing with URLs like https://app.assembla.com/login that have a redirect, I noticed that the old `HttpURLConnection` API automatically followed redirects but the new Java 11 HTTP client API was not. To ease migration I figured it would be best to follow redirects by default. I could not think of a case where you would not want to follow them, and it matches the behavior of the old API. Of course anyone who wants to opt out can always do so by creating a custom builder without this option (something that was not possible with the old API).